### PR TITLE
Add PlainTV and KindedTV patterns

### PR DESCRIPTION
**Pull request by zliu41**

### CODE REVIEW

File: th/Categorifier/TH.hs [hunk 1]
```diff
@@ -37,14 +40,30 @@ import Data.Map (Map)
...
+
+{-# COMPLETE PlainTV, KindedTV #-}
 #else
 type TyVarBndr _flag = TH.TyVarBndr
+
+pattern PlainTV :: TH.Name -> flag -> TyVarBndr flag
+pattern PlainTV n _flag = TH.PlainTV n
+
+pattern KindedTV :: TH.Name -> flag -> TH.Kind -> TyVarBndr flag
+pattern KindedTV n _flag k = TH.KindedTV n k
```
Comment (sellout) [inline th/Categorifier/TH.hs:64]: We don't need these patterns to be bidirectional, do we? They can be just
```haskell
pattern KindedTV n _flag k -> TH.KindedTV n k
```
But we can leave the 2.17 ones bidirectional.

File: th/Categorifier/TH.hs [hunk 2]
```diff
@@ -37,14 +40,30 @@ import Data.Map (Map)
...
 #if MIN_VERSION_template_haskell(2, 17, 0)
 type TyVarBndr flag = TH.TyVarBndr flag
+
+pattern PlainTV :: TH.Name -> flag -> TyVarBndr flag
+pattern PlainTV n flag = TH.PlainTV n flag
+
+pattern KindedTV :: TH.Name -> flag -> TH.Kind -> TyVarBndr flag
+pattern KindedTV n flag k = TH.KindedTV n flag k
+
+{-# COMPLETE PlainTV, KindedTV #-}
```
Comment (sellout) [inline th/Categorifier/TH.hs:56]: Can we pull the declarations and the `complete` pragma out of the conditionalizations?

**Review by sellout — CHANGES_REQUESTED** [review]

File: th/Categorifier/TH.hs (hunk 1 above)
Comment (sellout) [reply th/Categorifier/TH.hs:64]: Oh, nevermind ... I got the unidirectional ones backward. I guess they would be like
```haskell
pattern KindedTV :: TH.Name -> TH.Kind -> TyVarBndr flag
#if MIN_VERSION_template_haskell(2, 17, 0)
pattern KindedTV n k <- TH.KindedTV n _flag k
#else
pattern KindedTV n k = TH.KindedTV n k
#endif
```
where the _old_ can be bidirectional, but the new can't.

Comment (zliu41) [reply th/Categorifier/TH.hs:64]: I was hoping `pattern KindedTV :: TH.Name -> flag -> TH.Kind -> TyVarBndr flag` for both versions, but I'm not sure if that's possible. I think your approach is probably OK, so I'll do it

**Review by sellout — APPROVED** [review]

**Pull request merged**